plugin/tooling.py

```diff
@@ -535,3 +535,19 @@ def on_transport_close(self, exit_code: int, exception: Optional[Exception]) ->
         self._transport = None
         output = str(exception) if exception else '\n'.join(self._stderr_lines).rstrip()
         sublime.set_timeout(lambda: self._on_close(self._resolved_command, output, exit_code))
+
+
+class LspDoubleClickCommand(sublime_plugin.TextCommand):
+    click_count = 0
+
+    def run(self, edit: sublime.Edit, command: str, args: Dict[Any, Any]) -> None:
+        self.click_count += 1
+        if self.click_count == 2:
+            self.view.run_command(command, args)
+            self.click_count = 0
+            return
+        sublime.set_timeout(self.reset, 500)
+
+    @classmethod
```

Comment:
I didn't mean to use `@classmethod` if that's due to my comment. It should be a normal method of the instance since `run` changes instance property so this should too.

Reply:
I would also make `click_count` to be an instance variable, because a TextCommand will be created for each view, and if it is a class variable, all instances will share this variable now.

So in theory, you can click on an inlay hint in a view, and within half a second click on another hint in a different view, and it will still be counted as a double-click.

To be very precise, this would also still happen if you use an instance variable for this, but when clicking on different inlay hints in the same view. If this should be avoided, this command could check whether the given args are the same between `click_count == 0` and `click_count == 1` (i.e. temporarily store the `args` when the first click happens).

Reply:
> I would also make `click_count` to be an instance variable, because a TextCommand will be created for each view, and if it is a class variable, all instances will share this variable now.

In practice it should actually work fine because the class property will only be used for the initial value and if something sets self.click_count then it will also create new property on the instance that will shadow the class property.

That is as long as this `@classmethod` is removed.

Reply:
> To be very precise, this would also still happen if you use an instance variable for this, but when clicking on different inlay hints in the same view. If this should be avoided, this command could check whether the given args are the same between click_count == 0 and click_count == 1 (i.e. temporarily store the args when the first click happens).

Pushed some code that will prevent this.

Reply:
> I didn't mean to use @classmethod if that's due to my comment. It should be a normal method of the instance since run changes instance property so this should too.

Yes, that is how I (miss)understood it.